homeassistant/components/deako/light.py

```diff
@@ -0,0 +1,101 @@
+"""Binary sensor platform for integration_blueprint."""
+
+import logging
+from typing import Any
+
+from pydeako.deako import Deako
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+# Model names
+MODEL_SMART = "smart"
+MODEL_DIMMER = "dimmer"
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure the platform."""
+    client: Deako = config.runtime_data
+
+    devices = client.get_devices()
+    add_entities([DeakoLightEntity(client, uuid) for uuid in devices])
+
+
+class DeakoLightEntity(LightEntity):
+    """Deako LightEntity class."""
+
+    _attr_supported_color_modes: set[ColorMode]
```

Comment:
This was so that I didn't have to do a type check to use the `in` operator because I knew it was going to be a set, but I think this was a mistake. No need to overwrite the type, good call out.